pkg/ddl/backfilling_operators.go

```diff
@@ -301,18 +340,43 @@ func (src *TableScanTaskSource) Open() error {
 	return nil
 }
 
+// adjustRange adjusts the range so that we can skip the ranges that have been processed
+// according to the information of checkpoint manager.
+func (src *TableScanTaskSource) adjustStartKey(start, end kv.Key) kv.Key {
+	if src.cpMgr == nil {
+		return start
+	}
+	cpKey := src.cpMgr.LastProcessedKey()
+	if len(cpKey) == 0 {
+		return start
+	}
+	cpKeyNext := cpKey.Next()
+	if cpKeyNext.Cmp(start) > 0 || cpKey.Cmp(end) >= 0 {
```

Comment:
if `start <= end` and `cpKey >= end`, then `cpKeyNext > cpKey >= end >= start` -> `cpKeyNext > start`, so seems we only need to check `cpKeyNext.Cmp(start) > 0`

Reply:
It should be `cpKey < start` or `cpKey > end`... I will add a test for this in next PR.